File: ruby/difference-of-squares/difference_of_squares.rb
# :nodoc
class Squares
  def initialize(n)
    @n = n
  end

  def square_of_sum
    ((1..@n).inject(0) { |a, e| a + e })**2
  end

  def sum_of_squares
    (1..@n).inject(0) { |a, e| a + e**2 }
  end

  def difference
    square_of_sum - sum_of_squares
  end
end

File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(n):
    return reduce(lambda x, y: x + y, range(1, n + 1))**2


def sum_of_squares(n):
    return reduce(lambda x, y: x + y**2, range(1, n + 1))


def difference(n):
    return square_of_sum(n) - sum_of_squares(n)
